File: backend/src/main/java/appaanjanda/snooping/global/error/response/ErrorResponse.java
package appaanjanda.snooping.global.error.response;

import lombok.Builder;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
@Builder
public class ErrorResponse {
    private int httpStatus;
    private String errorCode;
    private String errorMessage;

    public static ErrorResponse of(HttpStatus httpStatus, String errorCode, String errorMessage) {
        return ErrorResponse.builder()
                .httpStatus(httpStatus.value())
                .errorCode(errorCode)
                .errorMessage(errorMessage)
                .build();
    }
}

File: android/data/src/main/java/com/appa/snoop/data/model/error/response/ErrorResponse.kt
package com.appa.snoop.data.model.error.response


import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import android.os.Parcelable

@Parcelize
data class ErrorResponse(
    @SerializedName("errorCode")
    val errorCode: String,
    @SerializedName("errorMessage")
    val errorMessage: String,
    @SerializedName("httpStatus")
    val httpStatus: Int
) : Parcelable